cmd/commands/root.go

```diff
@@ -0,0 +1,79 @@
+package command
+
+import (
+	"context"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	rookclient "github.com/rook/rook/pkg/client/clientset/versioned"
+	"github.com/spf13/cobra"
+	k8s "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	KubeConfig           string
+	KubeContext          string
+	OperatorNamespace    string
+	CephClusterNamespace string
+)
+
+// RootCmd represents the odf command
+var RootCmd = &cobra.Command{
+	Use:              "odf",
+	Short:            "Management and troubleshooting tools for ODF clusters.",
+	Aliases:          []string{"fdf"},
+	TraverseChildren: true,
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		if CephClusterNamespace != "" && OperatorNamespace == "" {
+			OperatorNamespace = CephClusterNamespace
+		}
+	},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	cobra.CheckErr(RootCmd.Execute())
+}
+
+func init() {
+	RootCmd.PersistentFlags().StringVar(&KubeConfig, "kubeconfig", "", "Openshift config path")
+	RootCmd.PersistentFlags().StringVar(&OperatorNamespace, "operator-namespace", "", "Openshift namespace where ODF operator is running")
+	RootCmd.PersistentFlags().StringVarP(&CephClusterNamespace, "namespace", "n", "openshift-storage", "Openshift namespace where StorageCluster is created")
+	RootCmd.PersistentFlags().StringVar(&KubeContext, "context", "", "Openshift context to use")
+}
+
+func GetClientsets(ctx context.Context) *k8sutil.Clientsets {
+	var err error
+
+	clientsets := &k8sutil.Clientsets{}
```

Comment:
Can we just call the `k8sutil.GetClientsets()` method? Why do we need to duplicate the method here? We're anyway using that `Clientsets` struct.